app/models/document.rb

```diff
@@ -39,6 +39,49 @@ def body
     details["body"] && details["body"].html_safe
   end
 
+  def intro
+    return nil unless body
+
+    intro = Nokogiri::HTML::DocumentFragment.parse(body)
+
+    # Strip all content following and including the first heading level specified by collapse_path
+    intro.css("h#{collapse_depth}").xpath("following-sibling::*").remove
+    intro.css("h#{collapse_depth}").remove
+
+    intro
+  end
+
+  def main
+    return nil unless body
+
+    document = Nokogiri::HTML::DocumentFragment.parse(body)
+    main = []
+    
+    # Identifies all headings of the level specified in collapse_path and creates an array of objects from the heading and its proceeding content up to the next heading
+    # This is so that it can be consumed by accordion components in the template
+    # See _manual_section.html.erb for how this is being rendered
+    document.css("h#{collapse_depth}").each do |heading|
```

Comment:
This could be a map:

```ruby
main = document.css("h#{collapse_depth}").map do |heading|
  # ...
  {
    heading: ...
    content: ...
  }
end
```

And rather than defining `main` and then returning it, we can just return the map directly:

```diff
- main = document.css("h#{collapse_depth}").map do |heading|
+ document.css("h#{collapse_depth}").map do |heading|
```

Reply:
Done!